src/util-thash.c

```diff
@@ -517,6 +524,10 @@ static THashData *THashDataGetNew(THashTableContext *ctx, void *data)
     // setup the data
     BUG_ON(ctx->config.DataSet(h->data, data) != 0);
 
+    if (ctx->config.DataLength) {
```

Comment:
can this move into `THashDataAlloc`?

Reply:
No. I tried that but found that alloc happens before the data is set (values filled in the respective data types) and it leads to incorrect results. We only have a length after the `DataSet` callback is done.